### PR TITLE
feat(cactus-web): Component theme changes phase one

### PR DESCRIPTION
**Pull request by NicolasSimmonds**
[CACTUS-244]

[CACTUS-244]: https://repayonline.atlassian.net/browse/CACTUS-244

### CODE REVIEW

**Comment by Dhalton** [comment]
I'm going to hold off on finishing my review on this until you make the changes we talked about for the MenuButton. Just to recap those for you, we wanted to handle that component in the following ways:

- Make the outline around the button for the focus state match the shape of the button
- If box shadows are turned off, we add a border to the dropdown and use the `lightContrast` color from the theme, and set the border width to the border size from the theme.

I opened my PR for the form elements, so if you want to check out what I did for the DateInput feel free to have a look at #228

File: modules/cactus-web/src/MenuButton/MenuButton.tsx [hunk 1]
```diff
@@ -88,9 +144,10 @@ type MenuButtonType = StyledComponent<typeof MenuButtonBase, any, {}, never> & {
 const MenuButton = styled(MenuButtonBase)`
   position: relative;
   box-sizing: border-box;
-  border-radius: 20px;
+  ${p => getShape(p.theme.shape)};
+  ${p => getBorder(p.theme.border)};
+  box-shadow: ${p => getBoxShadow(p.theme)};
```
Comment (Dhalton) [inline modules/cactus-web/src/MenuButton/MenuButton.tsx:149]: There was no box shadow on the button itself before; only on the dropdown, so let's remove this

File: modules/cactus-web/src/TextButton/TextButton.tsx [hunk 2]
```diff
@@ -89,7 +100,8 @@ export const TextButton = styled.button<TextButtonProps>`
       width: 100%;
       top: 0px;
       left: 0px;
-      border: 2px solid ${p => p.theme.colors.callToAction};
+      ${p => getBorder(p.theme.border)};
+      border-color: ${p => p.theme.colors.callToAction};
       border-radius: 20px;
```
Comment (Dhalton) [inline modules/cactus-web/src/TextButton/TextButton.tsx:105]: I wonder if we want to apply the shape from the theme here too. I think we should. Maybe check with Mariana/Hector before doing that

Comment (mikesoltow) [reply modules/cactus-web/src/TextButton/TextButton.tsx:105]: I think we should too, but it doesn't hurt to check.

File: modules/cactus-web/src/Button/Button.tsx [hunk 3]
```diff
@@ -158,9 +182,9 @@ export const Button = styled(ButtonBase)`
       width: calc(100% + 10px);
       top: -5px;
       left: -5px;
-      border: 2px solid ${p => p.theme.colors.callToAction};
-      border-radius: 20px;
+      border-color: ${p => p.theme.colors.callToAction};
       box-sizing: border-box;
+      ${p => getShape(p.theme.shape)}
```
[the remark this answers is not in the document]
Comment (Dhalton) [reply modules/cactus-web/src/Button/Button.tsx]: Nice catch, I didn't notice this

File: modules/cactus-web/src/Button/Button.test.tsx [hunk 4]
```diff
@@ -162,4 +163,27 @@ describe('component: Button', () => {
     fireEvent.click(getByTestId('not-clicked'))
     expect(onClick).not.toHaveBeenCalled()
   })
+  test
+})
+
+describe('With theme changes ', () => {
+  test('should have 2px border', () => {
+    const theme = generateTheme({ primaryHue: 200, border: 'thick' })
```
Comment (Dhalton) [inline modules/cactus-web/src/Button/Button.test.tsx:171]: The border defaults to `thick`, so I'd argue that this is already inadvertently being tested in other snapshots. Let's cover the bases a little more by testing `thin` borders instead

File: modules/cactus-web/src/MenuButton/MenuButton.test.tsx [hunk 5]
```diff
@@ -74,3 +75,47 @@ describe('component: MenuButton', () => {
     })
   })
 })
+
+describe('With theme changes ', () => {
+  test('Should have rounded borders', () => {
```
Comment (Dhalton) [inline modules/cactus-web/src/MenuButton/MenuButton.test.tsx:80]: Same thing goes for the three new tests in this file too...the shape defaults to `round`, border defaults to `thick`, and `boxShadows` is turned on by default, so these aren't really testing anything new. It's more important to test that the new styles are being applied

**Pull request merged**